cloudferrylib/os/compute/server_groups.py

```diff
@@ -192,6 +207,33 @@ def _deploy_server_group(self, server_group):
 
         return server_group
 
+    def get_server_group_id_by_vm(self, instance_id, instance_tenant):
+        """
+        Get Nova Server Group by it's member
+
+        :param instance_id: VM's ID
+        :param instance_tenant: VM's tenant name
+
+        :return str: Nova Server Group ID
+        """
+
+        client_config = copy.deepcopy(self.config)
+        client_config.cloud.tenant = instance_tenant
+
+        with keystone.AddAdminUserToNonAdminTenant(
+                self.identity.keystone_client,
+                client_config.cloud.user,
+                instance_tenant):
+            nclient = self.compute.get_client(client_config)
```

Comment:
`get_client()` signature sucks, would be much better to have explicit arguments here, instead of whole config. Not for this patch though